lib/mp4/captions-parser.js

```diff
@@ -0,0 +1,344 @@
+  * @return {Object.<Number, Object[]>} A mapping of video trackId to
+  *   a list of seiNals found in that track
+ **/
+var parseCaptionNals = function(segment, videoTrackIds) {
+  // To get the samples
+  var trafs = probe.findBox(segment, ['moof', 'traf']);
+  // To get SEI NAL units
+  var mdats = probe.findBox(segment, ['mdat']);
+  var captionNals = {};
+  var trafMdatPairs = [];
+
+  // Pair up each traf with a mdat
+  trafs.forEach(function(traf, index) {
+    var matchingMdat = mdats[index];
+    trafMdatPairs.push({
+      mdat: matchingMdat,
+      traf: traf
+    });
+  });
+
+  trafMdatPairs.forEach(function(pair) {
+    var mdat = pair.mdat;
+    var traf = pair.traf;
+    var tfhd = probe.findBox(traf, ['tfhd']);
+    // Exactly 1 tfhd per traf
+    var headerInfo = inspect.parseTfhd(tfhd[0]);
+    var trackId = headerInfo.trackId;
+    var tfdt = probe.findBox(traf, ['tfdt']);
+    // Either 0 or 1 tfdt per traf
+    var baseMediaDecodeTime = (tfdt.length > 0) ? inspect.parseTfdt(tfdt[0]).baseMediaDecodeTime : 0;
+    var truns = probe.findBox(traf, ['trun']);
+    var samples;
+    var seiNals;
+
+    // Only parse video data
+    if (videoTrackIds.includes(trackId) && truns.length > 0) {
+      samples = parseSamples(truns, baseMediaDecodeTime, headerInfo);
+
+      samples.sort(function(a, b) {
+        // This shouldn't ever happen
+        if (a.dataOffset === b.dataOffset) {
+          return a.pts - b.pts;
+        }
+        return a.dataOffset - b.dataOffset;
+      });
+
+      seiNals = findSeiNals(mdat, samples, trackId);
+
+      if (!captionNals[trackId]) {
+        captionNals[trackId] = [];
+      }
+
+      captionNals[trackId] = captionNals[trackId].concat(seiNals);
+    }
+  });
+
+  return captionNals;
+};
+
+/**
+  * Parses out inband captions from an MP4 container and returns
+  * caption objects that can be used by WebVTT and the TextTrack API.
+  * @see https://developer.mozilla.org/en-US/docs/Web/API/VTTCue
+  * @see https://developer.mozilla.org/en-US/docs/Web/API/TextTrack
+  *
+  * @param {Uint8Array} init - The init segment
+  * @param {Uint8Array} segment - The fmp4 segment containing imbedded captions
+  * @return {?Object[]} parsedCaptions - A list of captions or null if no video tracks
+  * @return {Number} parsedCaptions[].startTime - The time to show the caption in seconds
+  * @return {Number} parsedCaptions[].endTime - The time to stop showing the caption in seconds
+  * @return {String} parsedCaptions[].text - The visible content of the caption
+ **/
+var parseEmbeddedCaptions = function(init, segment) {
+  var captionStreams = {};
+  var parsedCaptions = [];
+
+  var videoTrackIds = getVideoTrackIds(init);
+  var captionStream = new CaptionStream();
+  var seiNals;
+  var timescales;
+  var trackId;
+  var trackTimescale;
+  var trackSeiNals;
+
+  if (videoTrackIds.length === 0) {
+    return null;
+  }
+
+  seiNals = parseCaptionNals(segment, videoTrackIds);
```

Comment:
If you're only using the first video track, you might as well only pass along that one video track here.

Reply:
Good point